doc/ztp/ztp_hld.md

```diff
@@ -0,0 +1,166 @@
+
+### Updategraph
+Needs to be updated to provide default config to the switch when ztp_enabled is true
+Needs to continue as currently designed when ztp_enabled is true and post_install is true
+Can be used to download config file and apply config
+
+### Interruption of ZTP
+Through command line utility “ZTP Disable” 
+Sets ZTP $enabled flag to FALSE
+If ZTP is interrupted mid-service, output is shown
+If ZTP service receives “NA_ZTP” flag, continue updategraph with default config
+ZTP service needs to have separate option for interruption than updategraph
+If ZTP service is interrupted, updategraph should still be enabled and allowed to continue
+
+### Syslog
+ZTP status should be logged to syslog server and /var/log/syslog
+
+### Build Option
+Need to allow build time option to compile with ZTP enabled
+
+# Error Cases
+- Case: Switch receives invalid SONiC image URL from DHCP server
+    - Switch should output error logs to syslog and apply default config.  Kill the ZTP service with aborted status, and allow other services to come up
+- Case: Switch receives invalid configuration URL from DHCP server
+    - Switch should output error logs to syslog and apply default config.  Kill the ZTP service with aborted status, and allow other services to come up
+- Case: Switch receives invalid post-config validation URL from DHCP server
+    - Switch should output error logs to syslog.  Disable the ZTP service, and allow other services to come up
+- Case: Switch receives invalid OS image 
+    - Switch should output error logs to syslog and apply default config.  Kill the ZTP service, and allow other services to come up
+- Case: Switch does not have enough memory to store the image
+    - Switch should output error logs to syslog and apply default config.  Kill the ZTP service, and allow other services to come up
+- Case: Switch receives invalid configuration from DHCP server
+    - Switch should output error logs to syslog, and gracefully exit the ZTP service, logging that image install completed successfully, and configuration script is invalid.
+- Case: Post-validation script fails
+    - The success on the script is processed by a remote server.  ZTP service is exited gracefully, logging with image install completed successfully.
+ 
+
+# ZTP CLI
+The ZTP CLI will allow the user to manually start, stop, and obtain status of the ZTP service.
+
+Will show the user whether ZTD is enabled, the date of the last execution of the ZTP script, and the completion status of the ZTP service
+
+    show ztp-status
+The ZTP status will show the user the current SONiC image, if the configuration was successful, and if the post-configuration script was run, as well as the time of the last successful completion of the ZTP service
+
+    config ztp enable
+The user can re-enable ZTP after a failed state using the CLI.  This puts the switch into “factory setting” and reloads with ZTP enabled, allowing for provisioning restart.
```

Comment:
This is a nit (so feel free to ignore!), but is this usage really limited to failure cases? Are there not cases where the user wants to re-run ZTP for other reasons (e.g. re-do a configuration)?